### PR TITLE
Uses ingestDataStream when its a new data stream only

### PR DESCRIPTION
**Pull request by pghorpade**
Connected to [ISLAND-811](https://jira.library.ucla.edu/browse/ISLAND-811)
Modified ucla_migration.drush.inc

### CODE REVIEW

File: ucla_migration.drush.inc [hunk 1]
```diff
@@ -3858,18 +3885,33 @@ EOQ;
...
+        else {
             $datastream = $object->constructDatastream('MODS', 'M');
             $datastream->label = "MODS Record";
             $datastream->mimetype = "text/xml";
+            $is_new_mods_ds = TRUE;
+        }
+
+        try {
+            $mods_file = file_get_contents($mods_path);
+            $datastream->content($mods_file);
```
Comment (IAMlKeno) [inline ucla_migration.drush.inc]: content is not a function here

File: ucla_migration.drush.inc [hunk 2]
```diff
@@ -3830,18 +3843,32 @@ EOQ;
...
-        $tei_file = file_get_contents($tei_path);
-        $datastream->content = $tei_file;
-        $object->ingestDatastream($datastream);
+        try {
+            $tei_file = file_get_contents($tei_path);
+            $datastream->content = $tei_file;
+            if ($is_new_mods_ds) {
+                $object->ingestDatastream($datastream);
+            }
+        }catch (Exception $e) {
```
Comment (IAMlKeno) [inline ucla_migration.drush.inc]: a nit pick, but the formatting is a little off in some places

File: ucla_migration.drush.inc [hunk 3]
```diff
@@ -3830,18 +3843,32 @@ EOQ;
...
+            $is_new_mods_ds = TRUE;
         }
         watchdog('Ingest Success', 'Found datastream @file_name', array('@file_name' => $tei_path), WATCHDOG_NOTICE);
         //$datastream->setContentFromFile($mods_path);
-        $tei_file = file_get_contents($tei_path);
-        $datastream->content = $tei_file;
-        $object->ingestDatastream($datastream);
+        try {
+            $tei_file = file_get_contents($tei_path);
+            $datastream->content = $tei_file;
```
Comment (IAMlKeno) [inline ucla_migration.drush.inc]: line 3857-3859 can be replaced with the datastream function setContentFromFile. i.e. `$datastream->setContentFromFile($tei_path)`

**Review by IAMlKeno — CHANGES_REQUESTED** [review]

File: ucla_migration.drush.inc (hunk 1 above)
Comment (pghorpade) [reply ucla_migration.drush.inc]: Good catch, this is fixed.

File: ucla_migration.drush.inc (hunk 2 above)
Comment (pghorpade) [reply ucla_migration.drush.inc]: Fixed formatting, for the catch block.

File: ucla_migration.drush.inc (hunk 3 above)
Comment (pghorpade) [reply ucla_migration.drush.inc]: Fixed it.

File: ucla_migration.drush.inc [hunk 4]
```diff
@@ -3818,30 +3830,44 @@ EOQ;
...
             $datastream->mimetype = "text/xml";
+            $is_new_tei_ds = TRUE;
         }
         watchdog('Ingest Success', 'Found datastream @file_name', array('@file_name' => $tei_path), WATCHDOG_NOTICE);
-        //$datastream->setContentFromFile($mods_path);
-        $tei_file = file_get_contents($tei_path);
-        $datastream->content = $tei_file;
-        $object->ingestDatastream($datastream);
+        try {
+            $datastream->setContentFromFile($tei_path)
```
Comment (IAMlKeno) [inline ucla_migration.drush.inc]: missing a semi-colon

**Review by IAMlKeno — CHANGES_REQUESTED** [review]

File: ucla_migration.drush.inc (hunk 4 above)
Comment (pghorpade) [reply ucla_migration.drush.inc]: Done

**Pull request merged**